src/node_file.h

```diff
@@ -21,10 +21,43 @@ using v8::Value;
 
 namespace fs {
 
+// structure used to store state during a complex operation, e.g., mkdirp.
+class FSContinuationData : public MemoryRetainer {
+ public:
+  FSContinuationData(uv_fs_t* req, int mode, uv_fs_cb done_cb)
+      : req(req), mode(mode), done_cb(done_cb) {
+  }
+
+  uv_fs_t* req;
+  int mode;
+  std::vector<std::string> paths;
+
+  void PushPath(const std::string& path) {
+    paths.push_back(path);
+  }
```

Comment:
I think everywhere where this is called, the `path` parameter is not used afterwards in the caller. Currently, this creates a copy of `path` and then disposes the original – it might make sense to use `std::string&&` as the parameter and move the string rather than copying it? (i.e. use `paths.emplace_back(std::move(path));` here and `continuation_data->PushPath(std::move(str));` elsewhere)